packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1487,6 +1496,42 @@ export class SnapController extends BaseController<
         return this.#unblockSnap(snapId as SnapId);
       }),
     );
+
+    if (!this.#featureFlags.autoUpdatePreinstalledSnaps) {
+      return;
+    }
+
+    const preinstalledVersionRange = '*' as SemVerRange;
+
+    await Promise.allSettled(
+      Object.values(this.state.snaps)
+        .filter((snap) => snap.preinstalled)
+        .map(async (snap) => {
+          const resolvedVersion = await this.#resolveAllowlistVersion(
+            snap.id,
+            preinstalledVersionRange,
+          );
+
+          if (
+            resolvedVersion !== preinstalledVersionRange &&
+            gtVersion(resolvedVersion as unknown as SemVerVersion, snap.version)
```

Comment:
Maybe `#resolveAllowlistVersion` should just return a `SemVerVersion`?

Reply:
The same function is used for resolving versions for regular Snaps on the registry, not sure it is worth it to mess with that. It can currently return a range or a version depending on what it finds in the registry.